pkg/services/sqlstore/user.go

```diff
@@ -199,7 +200,7 @@ func GetUserByLogin(query *m.GetUserByLoginQuery) error {
 
 	if err != nil {
 		return err
-	} else if !has {
+	} else if !has || user.IsDisabled {
```

Comment:
GetUserByLogin is used to get a user based on its login, not to login the user. We want this to still work even if the user has been disabled.

https://github.com/grafana/grafana/blob/b426ff5292e3aee77d1e07f38a84e6ff372f1955/pkg/login/auth.go#L28-L60

We need to go from this piece of code and implement the disabled logic in the code that calls the query, instead of into the query itself. Alternatively, we could implement a new query in sqlstore that does what GetUserByLogin does, but also reacts on the disabled state.

And, as we talked about yesterday in the enterprise weekly, I think the fact that the user is disabled should "leak" out to the user.

Reply:
Yes, will go this way.